### PR TITLE
CMake >= 3.23

### PR DESCRIPTION
**Pull request by wrtobin**
- [ ] Rebuild TPL docker containers
- [ ] Modify nightly runs `module load`s

Available on:
- [x] Quartz
- [x] Lassen
- [x] Crusher/Frontier
- [x] Tioga
- [x] Pangea2
- [x] Peacan
- [x] Perlmutter
- [ ] Others?

### CODE REVIEW

**Comment by TotoGaz** [comment]
@XL64 do we have `cmake` 3.23 on Pangea2?
@jhuang2601 Same question for our envs in Houston?

**Comment by jhuang2601** [comment]
> @XL64 do we have `cmake` 3.23 on Pangea2? @jhuang2601 Same question for our envs in Houston?

@TotoGaz Yes, this is the latest version `cmake/3.23.2` on Houston HPC

**Comment by untereiner** [comment]
@TotoGaz I asked for a cmake between 3.18 and 3.21 on Pangea2… the process is ongoing but I could update the requirements for cmake 3.23 minimum

**Comment by jhuang2601** [comment]
@drmichaeltcvx @paveltomin Any input for cmake version on CVX machine?

**Comment by victorapm** [comment]
Perlmutter supports cmake/3.24.3

**Comment by wrtobin** [comment]
@bmhan12 Can you comment on what / whether we need to address in the CI / nightly gitlab runs for this?

**Comment by bmhan12** [comment]
> @bmhan12 Can you comment on what / whether we need to address in the CI / nightly gitlab runs for this?

The nightly CI would need to `module load` a newer CMake version, straightforward.

The thirdPartyLibs Docker containers will nave to be tested and rebuilt:
https://github.com/search?q=repo%3AGEOS-DEV%2FthirdPartyLibs++cmake&type=code&p=1

Many configs are using older than 3.23

**Comment by untereiner** [comment]
Pangea2 is ok [cmake/3.23.2]

**Review by rrsettgast — APPROVED** [review]

**Review by drmichaeltcvx — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-cmake_minimum_required( VERSION 3.9 )
+cmake_minimum_required( VERSION 3.23 )
```
Comment (TotoGaz) [inline src/CMakeLists.txt:2]: I guess we do not need the `if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")` branch check on line 5?

**Review by TotoGaz — APPROVED** [review]

**Comment by bmhan12** [comment]
> Rebuild TPL docker containers

On second thought, the Docker containers do need to be rebuilt with CMake 3.23>=.
The CI jobs will use whatever CMake version is specified in the Dockerfile.
The current CI job failures reflect the CMake version installed.

**Comment by TotoGaz** [comment]
At least the following docker images
- https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/Stanford/Dockerfile
- https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/TotalEnergies/Dockerfile
- https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/clang-cuda/Dockerfile
- https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/gcc-cuda/Dockerfile
- https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/gcc-ubuntu/Dockerfile

have a version which are too old.

Travis conveniently reflects this (e.g.):
```
CMake Error at CMakeLists.txt:2 (cmake_minimum_required):
  CMake 3.23 or higher is required.  You are running version 3.19.8
```

So yes we should update the TPLs to upgrade `cmake` accordingly.
I'd suggest spanning from `3.23` to `3.27` to increase our validations.

**Comment by francoishamon** [comment]
To make progress in the (long) merge queue, I am skipping this PR to merge the next ones in the queue. They are small and should not impact this one

**Comment by CusiniM** [comment]
We ended up doing this in https://github.com/GEOS-DEV/GEOS/pull/2496 so I think we can close this PR.

**Comment by wrtobin** [comment]
@CusiniM Thanks for taking care of the TPL updates!

**Pull request closed without merging**